### PR TITLE
Add brotli decoder and WOFF2 support.

### PR DESCRIPTION
**Pull request by bruvzg**
Adds brotli decoder and enables WOFF2 font support.

Note: Adding decoder increase binary size by about 100 KB, but converting all editor and default theme fonts to WOFF2 can decrease it by 4.4 MB. Tested by using https://github.com/google/woff2 on all included fonts, not included into this PR, since quality of conversion need to be tested first.

### CODE REVIEW

File: modules/freetype/SCsub [hunk 1]
```diff
@@ -57,6 +57,22 @@ if env["builtin_freetype"]:
...
+        "common/platform.c",
+        "common/transform.c",
+        "dec/bit_reader.c",
+        "dec/decode.c",
+        "dec/huffman.c",
+        "dec/state.c",
+    ]
+    thirdparty_sources += [thirdparty_brotli_dir + file for file in thirdparty_brotli_sources]
+    env_freetype.Append(CPPDEFINES=["FT_CONFIG_OPTION_USE_BROTLI"])
+    env_freetype.Prepend(CPPPATH=["#thirdparty/brotli/include"])
```
Comment (akien-mga) [inline modules/freetype/SCsub:74]:
```suggestion
    env_freetype.Prepend(CPPPATH=[thirdparty_brotli_dir + "include"])
```

File: thirdparty/README.md [hunk 2]
```diff
@@ -29,6 +29,18 @@ Files extracted from upstream source:
 - `LICENSE`
 
 
+## brotli
+
+- Upstream: https://github.com/google/brotli
+- Version: 1.0.9 (e61745a6b7add50d380cfd7d3883dd6c62fc2c71, 2020)
```
Comment (akien-mga) [inline thirdparty/README.md:35]: We could update to latest Git commit, it seems like there have been a number of significant fixes and no news of any plans to release a maintenance update.

File: COPYRIGHT.txt [hunk 3]
```diff
@@ -130,6 +130,11 @@ Comment: Basis Universal
 Copyright: 2019, Binomial LLC.
 License: Apache-2.0
 
+Files: ./thirdparty/brotli/
+Comment: Brotli
+Copyright: 2009, 2010, 2013-2016 by the Brotli Authors.
+License: MIT
```
Comment (akien-mga) [inline COPYRIGHT.txt:136]:
```suggestion
License: Expat
```

That's the name used by Debian in their spec, as MIT is often ambiguous (or _was_, nowadays I think everyone means the same with MIT, but well): https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/#license-short-name

**Comment by akien-mga** [comment]
Might be useful for the JavaScript platform @Faless @Calinou :)

**Comment by Calinou** [comment]
For reference, WOFF1 support was exposed in https://github.com/godotengine/godot/pull/52052.

This is definitely a good thing to have, especially since many web fonts are now distributed in a WOFF2 format only (and using it in another format requires manual conversion).

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**